### PR TITLE
DEPR: loc with listlikes with missing elements

### PR DESCRIPTION
**Pull request by jbrockmendel**
I'm less confident in this one than in others.  Some of the affected tests seem to involve non-missing labels in duplicate-containing indexes.  Did I conflate multiple issues/deprecations?

### CODE REVIEW

File: pandas/io/formats/excel.py [hunk 1]
```diff
@@ -393,16 +393,11 @@ def __init__(
...
-                warnings.warn(
-                    "Not all names specified in 'columns' are found; "
-                    "this will raise a KeyError in the future",
-                    FutureWarning,
-                )
+                # Deprecated in GH#17295, enforced in 1.0.0
+                raise KeyError("Not all names specified in 'columns' are found")
 
-            self.df = df.reindex(columns=cols)
+            self.df = df.reindex(columns=cols)  # TODO: Should this change?
```
Comment (jreback) [inline pandas/io/formats/excel.py]: yeah I think you can just remove this now (self.df = df)

File: pandas/tests/indexing/test_iloc.py [hunk 2]
```diff
@@ -750,20 +750,9 @@ def test_iloc_non_unique_indexing(self):
...
-            new_list.append(s * 2)
-            new_list.append(s * 3)
-
-        expected = DataFrame(new_list)
-        expected = concat([expected, DataFrame(index=idx[idx > sidx.max()])], sort=True)
-        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
-            result = df2.loc[idx]
-        tm.assert_frame_equal(result, expected, check_index_type=False)
+        with pytest.raises(KeyError, match="with any missing labels"):
+            # TODO: should non-uniquness play a part in the error message?
```
Comment (jreback) [inline pandas/tests/indexing/test_iloc.py]: hmm, yes I would agree here

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm, couple of comments

File: pandas/tests/indexing/test_iloc.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_iloc.py]: sounds good, ill see what i can do here.

BTW in #6581 are we removing things in the 0.25.0 section or only the earlier ones?

Comment (jbrockmendel) [reply pandas/tests/indexing/test_iloc.py]: hmm the non-uniqueness doesn't seem to play any part in us getting here, so im now leaning towards not having a special message

**Comment by jbrockmendel** [comment]
rebased+green

File: pandas/tests/series/indexing/test_numeric.py [hunk 3]
```diff
@@ -123,12 +123,10 @@ def test_get_nan_multiple():
     s = pd.Float64Index(range(10)).to_series()
 
     idx = [2, 30]
-    with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
-        tm.assert_series_equal(s.get(idx), Series([2, np.nan], index=idx))
+    assert s.get(idx) is None
```
Comment (jreback) [inline pandas/tests/series/indexing/test_numeric.py]: hmm, was this doing a reindex before?

Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_numeric.py]: AFAICT yes

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**